node/cli/src/command.rs

```diff
@@ -199,39 +251,68 @@ pub fn run() -> Result<()> {
 			})
 		},
 		Some(Subcommand::Benchmark(cmd)) => {
-			let runner = cli.create_runner(cmd)?;
-			// Switch on the concrete benchmark sub-command-
-			match cmd {
-				BenchmarkCmd::Pallet(cmd) =>
-					if cfg!(feature = "runtime-benchmarks") {
-						runner.sync_run(|config| cmd.run::<Block, TemplateRuntimeExecutor>(config))
-					} else {
-						Err("Benchmarking wasn't enabled when building the node. \
-					You can enable it with `--features runtime-benchmarks`."
-							.into())
+			if cfg!(feature = "runtime-benchmarks") {
```

Comment:
it seems like previously `runtime-benchmarks" check was only inside `BenchmarkCmd::Pallet(cmd)` condition but since you brought it out now all of these conditions need that feature , I'm not totally sure if that is necessary for other cases?

Reply:
Good point, yes we dont want to specify this feature flag non related cmds, i moved it inside again